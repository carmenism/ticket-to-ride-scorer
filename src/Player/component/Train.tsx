
import * as React from 'react';

export interface TrainProps {
  handleChange: (event: any) => void;
  length: number;
  name: string;
  value: number;
}

const PlayerForm: React.SFC<TrainProps> = (props) => {
  const trainMin: number = 0;
  const trainMax: number = 15;

  return (
    <label>
      {props.length}:
      <input
        type="number"
        name={props.name}
        value={props.value}
        onBlur={props.handleChange}
        onInput={props.handleChange}
        onChange={props.handleChange}
        min={trainMin}
        max={trainMax}
      />
    </label>
  );
};

export default PlayerForm;
