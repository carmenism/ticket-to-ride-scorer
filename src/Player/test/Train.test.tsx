import * as React from 'react';
import { shallow } from 'enzyme';
import Train from '../component/Train';

describe('Train', () => {
  let wrapper: any;
  const handleChange = jest.fn();
  const name: string = 'numberTrainsLength1';

  beforeEach(() => {
    wrapper = shallow(
      <Train
        handleChange={handleChange}
        length={1}
        name={name}
        value={4}
        key={name}
      />
    );
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
