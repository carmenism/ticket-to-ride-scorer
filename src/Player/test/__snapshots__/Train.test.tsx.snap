// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Train Matches snapshot 1`] = `
<label>
  1
  :
  <input
    max={15}
    min={0}
    name="numberTrainsLength1"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onInput={[MockFunction]}
    type="number"
    value={4}
  />
</label>
`;
